### DISCUSSION TITLE
implement triggers where action on one table leads to insertion in another using middlewares

### DISCUSSION THREAD

**Discussion by satya-nutella** [prisma/prisma#18653] [Q&A]
## Description

Since Prisma lacks support for soft deletes when it comes to complex queries (with `includes`), I am trying to implement a `archived table` method of achieving soft deletes.

The idea is to have the following flow:
When a model is deleted, rather than delete the model, move it to a table with `archive<tablename>` and then delete row from the original table. That way we still have reference to the original row in the archive table and `find` queries dont accidentally return the deleted values.

The way i was thinking of achieving this is to intercept the delete and deleteMany calls in the middleware. However, i am having issues in dynamically getting the right model in the middleware.

For example, if I am intercepting delete on a table named "user", I want to do something like:

```
if (params.model.startsWith('Archived') === false) {
   const result = await next(params)
   
   client[`archived${params.model}`].create({
     data: { ...result }
   })
}
```

Is there a way to achieve this.

I also looked into `prisma query extensions` but couldnt find anything there either

**Comment by nurul3101** [comment]
Hey @meehawk 👋 

Just to confirm, did you had a look at this [example](https://www.prisma.io/docs/concepts/components/prisma-client/middleware/soft-delete-middleware) of implementing soft delete? Would this not work in your use case?

**Reply by satya-nutella** [reply]
Hey @nurul3101 unfortunately it wont. I want to exclude deleted rows from our query results. This implementation doesnt work since it handles only for simple cases and quickly breaks as the queries get complex for example queries with includes.

Also couldnt find any third party/community maintained plugin that tackles this without caveats.

by actually deleting the row and just shifting logic to an archive table, we are able to bypass the issue

**Reply by nurul3101** [reply]
I am wondering if, instead of Middleware, you can use Prisma Client Extensions like [this](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/query#modify-a-specific-operation-in-all-models-of-your-schema) and modify the delete operation to insert in another table.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by satya-nutella** [comment]
@nurul3101 any update on this?
